.changelog/11257.txt

```diff
@@ -0,0 +1,7 @@
+```release-note:breaking-change
+config: `verify_incoming_rpc` now defaults to true if either `ca_file` or `ca_path` are set. Previously it defaulted to false in all cases.
+```
+
+```release-note:deprecation
+config: `verify_incoming` is deprecated and may be removed in a future version. Use `verify_incoming_https` and `verify_incoming_rpc` instead.
```

Comment:
I think this would break the ability for non TLS'ed clusters to enable TLS without downtime. See [this learn guide](https://learn.hashicorp.com/tutorials/consul/tls-encryption-secure-existing-datacenter#configure-the-servers) on how the online migration is performed. Basically it involves setting up the certs with all verification off and then once client agents etc. are all initiating TLS connections, then the verification settings can be turned on.

Changing the default when ca material is provided seems like a good idea, but removal of the ability to disable that verification while still using certs is going to make securing an existing system require downtime so I think we should avoid it.